### PR TITLE
[main] fix padding mask docstring

### LINKED ISSUE #5569: Inconsistent padding_mask docstring semantics in MoE router
**Describe the bug**

`@NVIDIA/mcore-oncall`

There appears to be an inconsistency between the `padding_mask` docstrings in `megatron/core/transformer/moe/router.py` and the actual MCore implementation.

The implementation consistently treats:

* `padding_mask=True` as padding / excluded token
* `padding_mask=False` as valid / included token

However, some docstrings in `router.py` describe `padding_mask` as indicating non-padding tokens, with `True for valid tokens, False for padding tokens`.

For example, `TopKRouter.routing()` currently documents:

> Boolean mask indicating non-padding tokens. Shape [seq_length, bsz]. True for valid tokens, False for padding tokens.

But the implementation and related utilities use the opposite semantics.

**Steps/Code to reproduce bug**

No runtime failure is required; this is a documentation/semantic inconsistency.

Relevant code links:

* Docstring says `True for valid tokens`:
  https://github.com/NVIDIA/Megatron-LM/blob/d0b3b775498a8ae6870075f1a5fc32bc260aa1b3/megatron/core/transformer/moe/router.py#L590-L597

* Expert-bias logic masks out `padding_mask=True`:
  https://github.com/NVIDIA/Megatron-LM/blob/d0b3b775498a8ae6870075f1a5fc32bc260aa1b3/megatron/core/transformer/moe/router.py#L573-L588

* `moe_utils.py::z_loss_func()` documents the intended semantics as `True = padding (exclude), False = valid (include)`:
  https://github.com/NVIDIA/Megatron-LM/blob/d0b3b775498a8ae6870075f1a5fc32bc260aa1b3/megatron/core/transformer/moe/moe_utils.py#L165-L169

* `compute_routing_scores_for_aux_loss()` also uses `valid_mask = (~padding_mask).unsqueeze(-1)`:
  https://github.com/NVIDIA/Megatron-LM/blob/d0b3b775498a8ae6870075f1a5fc32bc260aa1b3/megatron/core/transformer/moe/moe_utils.py#L892-L897

Minimal reproduction is just inspecting these definitions: the docstring says `True = valid`, while implementation uses `~padding_mask` to obtain valid tokens.

**Expected behavior**

The docstrings in `router.py` should match the implementation.

Suggested wording:

```python
padding_mask (torch.Tensor, optional): Boolean mask indicating padding positions.
                                       Shape [seq_length, bsz]. True for padding/excluded tokens,
                                       False for valid/included tokens. Defaults to None.
```

This likely applies to at least:

* `TopKRouter.routing()`
* `TopKRouter.apply_z_loss()`

**Additional context**

I do not think the MCore runtime mask semantics are wrong. The implementation appears consistent: `padding_mask=True` means padding/excluded. The issue seems limited to stale or inverted docstrings in `router.py`.

### PR DESCRIPTION
**Pull request by HaochenYuan**
- [x] I, the PR author, have personally reviewed every line of this PR.

# What does this PR do?

Fixes https://github.com/NVIDIA/Megatron-LM/issues/5569.

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact @NVIDIA/mcore-oncall.

## Issue tracking

For PRs from open-source community contributors:

- **New features**: a linked issue is **required**. Please open a [feature request](https://github.com/NVIDIA/Megatron-LM/issues/new?template=feature_request.md) and reference it here before submitting the PR.
- **Small updates (bug fixes, minor improvements)**: a linked issue is **recommended** and will accelerate the PR review process.

Linked issue: 

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment @NVIDIA/mcore-oncall to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

### CODE REVIEW

**Review by Victarry — APPROVED** [review]

**Review by Phlip79 — APPROVED** [review]

**Comment by HaochenYuan** [comment]
@NVIDIA/core-adlr @NVIDIA/mixture-of-experts-adlr

**Review by yashaswikarnati — APPROVED** [review]

File: tests/unit_tests/transformer/moe/test_moe_layer.py [hunk 1]
```diff
@@ -368,16 +368,17 @@ def test_moe_layer_recompute_forward_backward(
         )
 
         # Create padding mask if needed: shape [batch_size, sequence_length]
+        # Convention: True = padding (exclude), False = valid (include)
         padding_mask = None
         if with_padding_mask:
-            padding_mask = torch.ones(
+            padding_mask = torch.zeros(
```
Comment (jaredcasper) [inline tests/unit_tests/transformer/moe/test_moe_layer.py:374]: How did this test pass before if it had the padding mask sense incorrect? Seems like the test must not be actual using the padding_mask then? Can we add a test that uses it?

Comment (HaochenYuan) [reply tests/unit_tests/transformer/moe/test_moe_layer.py:374]: We already have `test_padding_mask_removes_padding_tokens`, which compares a padded-and-masked input against the same valid tokens with the padding physically removed, and checks both aux loss and gradients.  
The padding mask here is passed into the existing recompute test. This test only checks forward/backward compatibility and does not validate the mask semantics since one test has done it.

**Review by jaredcasper — APPROVED** [review]

**Review by yaox12 — APPROVED** [review]
